Reference/Common-Pitfalls/index.md

```diff
@@ -0,0 +1,466 @@
+You could try to think about every query you make as an SQL call - you don't want to make too many otherwise the performance of 
+your website is going to suffer.
+
+Here's a common pitfall that is seen. Let's continue the menu example, in this example the menu is going to be rendered
+using the current page's root node::
+
+```
+<ul>
+    <li><a href="@Model.Site().Url">@Model.Site().Name</a></li>
+    @foreach(var node in Model.Site().Children) 
+    {
+        <li><a href="@node.Url">@node.Name</a></li>
+    }
+</ul>
+```
+
+The syntax `@Model.Content.Site()` is actually shorthand for doing this:
+`Model.Content.AncestorsOrSelf(1)` which means it is going to traverse up the tree until it reaches an ancestor node
+with a level of one. As mentioned above, traversing costs resources and in this example there is 3x traversals being done
+for the same value. Instead this can be rewritten as:
+
+```
+@{
+    var site = @Model.Content.Site();
+}
+<ul>
+    <li><a href="@site.Url">@site.Name</a></li>
+    @foreach(var node in site.Children) 
+    {
+        <li><a href="@node.Url">@node.Name</a></li>
+    }
+</ul>
+```
+
+## Dynamics
+
+In Umbraco version 8+ dynamic support for access to IPublishedContent will be removed. 
+There are a few reasons for this:
+
+* Dynamics are much slower than their strongly typed equivalent
+* The codebase for Dynamics is difficult to maintain and it's massive
+* Many querying concepts in Dynamics are difficult to understand and need to be memorize due to all of the string syntax required
+* It is much harder to debug and to know if there are errors since the syntax is not typed or compiled
+* No intellisense is possible inside Visual Studio
+* ModelsBuilder is part of the Umbraco Core and provides much nicer and strongly typed access to property accessors and querying in your views
+
+How do you know if you are using Dynamics?
+
+* If you are using `@CurrentPage` then __you are__ using dynamics
+* If you are using the UmbracoHelper query methods like `@Umbraco.Content` or `@Umbraco.Media` instead of the typed methods like `@Umbraco.TypedContent` and `@Umbraco.TypedMedia` then __you are__ using dynamics
+
+It is strongly advised that you use the strongly typed `@Model.Content` instead of `@CurrentPage` models in your views,  
+this will actually perform much better and you'll be forward compatible with Umbraco v8+ with regards to querying `IPublishedContent`. 
+
+A large problem with the performance of dynamics is having to parse string syntax such as:
+`@CurrentPage.Children.Where("DocumentTypeAlias == \"DatatypesFolder\" && Visible")` and turn that into something that is compilable when 
+instead it can just be written as something that compiles
+
+## Using the Services layer in your views
+
+The Services layer of Umbraco is for manipulating the business logic of Umbraco directly to/from the database. 
```

Comment:
I'll add this but as a side note, your examples above with `stilNope`, `better`, `evenBetter` are basically all the same. It really depends on what querying you are doing and what services you rely on - and then what you want to mock for unit tests. The `Umbraco.ContentQuery.TypedContent` is actually the preferred when speaking in terms of unit testability because `ContentQuery` is an interface which can be directly mocked whereas if you want to mock `UmbracoContext.ContentCache.GetById` or `UmbracoHelper.TypedContent` - these would be much more difficult to mock.